featurebyte/routes/lazy_app_container.py

```diff
@@ -214,7 +214,9 @@ def _get_key(self, key: str) -> Any:
             filtered_deps, self.app_container_config.get_class_def_mapping()
         )
         new_deps = build_deps(ordered_deps, self.instance_map)
-        self.instance_map.update(new_deps)
+        self.instance_map.update(
+            {name: self._handle_block_modification_check(dep) for name, dep in new_deps.items()}
+        )
```

Comment:
This is the main fix the for bug. In my previous PR, i missed this part & it causes some document service's block modification callback is not set properly.